packages-exp/auth-compat-exp/src/auth.ts

```diff
@@ -199,28 +200,29 @@ export class Auth
     );
   }
   async setPersistence(persistence: string): Promise<void> {
-    function convertPersistence(
-      auth: exp.Auth,
-      persistenceCompat: string
-    ): exp.Persistence {
-      _validatePersistenceArgument(auth, persistence);
-      switch (persistenceCompat) {
-        case Persistence.SESSION:
-          return exp.browserSessionPersistence;
-        case Persistence.LOCAL:
-          return isIndexedDBAvailable()
-            ? exp.indexedDBLocalPersistence
-            : exp.browserLocalPersistence;
-        case Persistence.NONE:
-          return exp.inMemoryPersistence;
-        default:
-          return exp._fail(exp.AuthErrorCode.ARGUMENT_ERROR, {
-            appName: auth.name
-          });
-      }
+    _validatePersistenceArgument(this.auth, persistence);
+    let converted;
+    switch (persistence) {
+      case Persistence.SESSION:
+        converted = exp.browserSessionPersistence;
+        break;
+      case Persistence.LOCAL:
+        // Not using isIndexedDBAvailable() since it only checks if indexedDB is defined.
+        const isIndexedDBFullySupported = await (exp.indexedDBLocalPersistence as exp.PersistenceInternal)._isAvailable();
```

Comment:
You'll need to run the prettier. From the root of the directory (after a commit on a clean client):
`node tools/gitHooks/precommit.js`

Reply:
Prettier insists on formatting it like this -- I tried wrapping the lines myself and `node tools/gitHooks/precommit.js` would undo it and make it all in one line.